onnxruntime/core/mlas/lib/sgemm.cpp

```diff
@@ -26,6 +32,24 @@ Module Name:
 
 #define MLAS_SGEMM_TRANSA_ROWS              12
 
+#if defined(USE_KLEIDIAI) && !defined(_MSVC_LANG)
+bool UseKleidiAISgemm(
+    CBLAS_TRANSPOSE TransA,
+    CBLAS_TRANSPOSE TransB,
+    size_t N, size_t K
+    )
+{
+    if (TransA == CblasNoTrans && TransB == CblasNoTrans && MLAS_CPUIDINFO::GetCPUIDInfo().HasArm_SME()) {
+        const size_t n_step = kai_get_n_step_matmul_clamp_f32_f32p2vlx1_f32p2vlx1biasf32_sme2_mopa();
+        const size_t BlockedN = (N + n_step - 1) / n_step;
+        const size_t AlignedN = BlockedN * n_step;
+
+        return AlignedN > 64 && K > 1;
```

Comment:
Maybe add some documentation on why this is the criteria to be met before using the Kleidi micro-kernels ?